src/layer/AbstractSentinelHubV1OrV2Layer.ts

```diff
@@ -7,22 +7,31 @@ import { GetMapParams, ApiType, PaginatedTiles } from 'src/layer/const';
 import { wmsGetMapUrl } from 'src/layer/wms';
 import { AbstractLayer } from 'src/layer/AbstractLayer';
 
+interface ConstructorParameters {
+  instanceId?: string | null;
+  layerId?: string | null;
+  evalscript?: string | null;
+  evalscriptUrl?: string | null;
+  title?: string | null;
+  description?: string | null;
```

Comment:
We should use "interface inheritance" ([docs](https://www.typescriptlang.org/docs/handbook/interfaces.html#extending-interfaces)) for the constructor parameters that are the same as in the super class / abstract class

Reply:
Not sure about that... In theory, it is not necessarily true that the parameters of a child class include all the options of the parent class (though currently this is so, so maybe this is a moot point). Also, that would mean that we would need to export all the interfaces.

Is there any reason why we should do it?

Reply:
> Is there any reason why we should do it?

I just have one: They are repeated ... a number of times and I am too lazy to check if some param from the `Abstract...Layer` classes is missing in the other `...Layer` classes :)

> Edit: I was actually expecting a question why I used `interface`s instead of `type`s... ;)

It looks like a nice solution so I don't have a reason to question the use of `interface`s

Reply:
I see your point, and I do agree to some extent. The problem is just that it seems conceptually wrong to me... For example, AbstractSH3Layer could just as easily be a subclass of WmsLayer (it is not, but for the sake of the argument - and there are reasons why it should be). In this case you would have `baseUrl` in the parent class, but only `instanceId` in the child class (`baseUrl` is constructed from dataset-specific intrinsic knowledge + `instanceId`).

I think it makes sense not to extend constructor parameters tbh... In any case, can we maybe leave it now as it is and then try to find agreement live?

Reply:
No problem, it can stay like that. Resolving.